deploy/gke/redaction/README.md

```diff
@@ -68,4 +68,105 @@ When you are done, you can clean up everything you've done with the following st
+
+### Prerequisites 
+You should be able to follow all steps mentioned in this README till [Verify the Deployment](#verify-the-deployment) successfully. This ensures that there are no permissions issues and you are able to succeesfully run a GKE cluster and connect to it. 
+
+### Source of telemetry data
+
+We would be using JSON files containing telemetry data in OTLP format as the source of telemetry data. A sample file has been provided here - [testdata.json](./otlp-data/testdata.json). The collector running in the cluster would read this file and treat the data in this file as if it were coming from a running application. 
+
+### Updating the collector configuration to receive data from JSON file
+
+Next, we would update our collector configuration file - [otel-config.yaml](./otel-config.yaml) to add a reciever that is able to receive telemetry data from the added JSON file. 
+
+Under the `receivers` section in the config file, add the following configuration - 
+
+```
+  otlpjsonfile:
+    start_at: "beginning"
+    include:
+      - "/mnt/testdata/*.json"
+
+```
+For more details about this particular receiver, check [otlpjsonfilereceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/otlpjsonfilereceiver).
+
+***NOTE:** The path in `include` points to where the file would be mounted within the kubernetes cluster environment and is therefore different from where it is present on your local machine. More information on this in [the next section](#making-the-json-file-available-within-the-cluster).*
+
+Next, add this receiver in the traces pipeline, so your trace pipeline looks like - 
+```
+ traces:
+      receivers: [otlp, otlpjsonfile]
+      processors: [memory_limiter, batch, resourcedetection/gke, redaction/credit_cards]
+      exporters: [googlecloud, logging]
+```
+ 
+### Making the JSON file available within the cluster 
+
+Simply adding the JSON file within the directory does not grant the running cluster the access to this file. In order to get access to this file within the kubernetes environment, we would be mounting this file as a kubernetes `configMap` for a new deployment - 
+
+1. Make sure that there are no current deployments by running `kubectl get deployments -n $OTEL_NAMESPACE`. If there are active deployments, delete them using `kubectl delete`. 
+2. Since we also updated our collector configuration, we will need to recreate the `otel-config` configMap too. Delete the old configMaps, if any before proceeding. 
+    - To delete configMaps, you can use `kubectl delete configmaps <configMap name> -n $OTEL_NAMESPACE`. 
+3. Recreate `otel-config` configMap with the updated configuration - `kubectl create configmap otel-config --from-file=./otel-config.yaml -n $OTEL_NAMESPACE`. 
```

Comment:
This section would flow better if each step the user has to do is broken into a code block (three ```) rather than inline (`), eg

3. Recreate `otel-config` configMap with the updated configuration 
```
`kubectl create configmap otel-config --from-file=./otel-config.yaml -n $OTEL_NAMESPACE`. 
```

(same for steps 4,6,7)